src/datatype.ts

```
@@ -37,14 +38,26 @@ export class Datatype {
   number(
     options?: number | { min?: number; max?: number; precision?: number }
   ): number {
-    const opts = typeof options === 'number' ? { max: options } : options ?? {};
+    let min: number;
+    let max: number;
+    let precision: number;
 
-    const min = typeof opts.min === 'number' ? opts.min : 0;
-    let max = typeof opts.max === 'number' ? opts.max : min + 99999;
-    const precision = typeof opts.precision === 'number' ? opts.precision : 1;
+    if (typeof options === 'object') {
+      min = options.min ?? 0;
```

The issue with this approach is that it does not check proper type of this property. I know we are using types, but they would only be checked if someone is using TS. When he is not, he could easily call this function as `faker.datatype.number({min: 'a'}). How should it work in such case? Throw an error? Assume option is invalid and take default? Right now it will just accept it and (I think) return `NaN`... Maybe that's fine, don't know. Before it checking for the type and ignoring invalid options, so in a way its a breaking change...

---

> what's the value of this refactor?

To align it to the #791 impl

> easily call this function as `faker.datatype.number({min: 'a'})`

If someone would do this, they just disrespect the API contract and we would need to check every single input parameter in every single function.
We don't do that. We do not want to do that.
If someone disrespect the API contract, they need to life with unsupported behavior.
This PR does not change the API contact / function signature.

---

Just my $.2 I preferred the old implementation. Felt simpler and shorter. #791 is new so it should rather align to what we had then vice versa :)

---

After having another look at this. I think I like the options remapping L40 (but maybe directly overwrite the options).

````ts
if (typeof options === number) {
 options = { ... }
}
````

The default should be set in the signature as well. That would remove the need for the fallback from that check too.
It would also explicitly document the default in our api docs.

After that we don't need the big if anymore either as well as allowing us to use const for the values again.
So that should hopefully solve most of @pkuczynski's issues with these changes.

However, I agree with @Shinigami92 that we don't have to check for most of out API contract.
If we do check, then we should explicitly throw an error instead of silently ignoring the wrong value.

So please revert this partially/change this to be closer to pkuczynski's implementation.